tests/test_unreachable.py

```diff
@@ -104,3 +104,223 @@ def foo():
         return False
 """)
     check_unreachable(v, 4, 1, 'return')
+
+
+def test_raise_assignemnt(v):
+    v.scan("""\
+def foo():
+    raise ValueError
+    li = []
+""")
+    check_unreachable(v, 3, 1, 'raise')
+
+
+def test_multiple_raise_statements(v):
+    v.scan("""\
+def foo():
+    a = 1
+    raise
+    raise KeyError
+    # a comment
+    b = 2
+    raise CustomDefinedError
+""")
+    check_unreachable(v, 4, 4, 'raise')
+
+
+def test_return_with_raise(v):
+    v.scan("""\
+def foo():
+    a = 1
+    return
+    raise ValueError
+    return
+""")
+    check_unreachable(v, 4, 2, 'return')
+
+
+def test_return_comment_only(v):
+    v.scan("""\
+def foo():
+    return None
+    # foo does nothing
+""")
+    assert v.unreachable_code == []
+
+
+def test_return_comment_and_code(v):
+    v.scan("""\
+def foo():
+    return
+    # This is a comment
+    print("Hello World")
+""")
+    check_unreachable(v, 4, 1, 'return')
+
+
+def test_raise_with_return(v):
+    v.scan("""\
+def foo():
+    a = 1
+    raise
+    return a
+""")
+    check_unreachable(v, 4, 1, 'raise')
+
+
+def test_raise_error_message(v):
+    v.scan("""\
+def foo():
+    raise SomeError("There is a problem")
+    print("I am unreachable")
+""")
+    check_unreachable(v, 3, 1, 'raise')
+
+
+def test_raise_try_except(v):
+    v.scan("""\
+def foo():
+    try:
+        a = 1
+        raise
+    except IOError as e:
+        print("We have some problem.")
+        raise
+        print(":-(")
+""")
+    check_unreachable(v, 8, 1, 'raise')
+
+
+def test_raise_with_comment_only(v):
+    v.scan("""\
+def foo():
+    raise
+    # A comment
+""")
+    assert v.unreachable_code == []
+
+
+def test_raise_with_comment_and_code(v):
+    v.scan("""\
+def foo():
+    raise
+    # This is a comment
+    print("Something")
+    return None
+""")
+    check_unreachable(v, 4, 2, 'raise')
+
+
+def test_continue_basic(v):
+    v.scan("""\
+def foo():
+    if bar():
+        a = 1
+    else:
+        continue
+        a = 2
+""")
+    check_unreachable(v, 6, 1, 'continue')
+
+
+def test_continue_one_liner(v):
+    v.scan("""\
+def foo():
+    for i in range(1, 10):
+        if i == 5: continue
+        print(1/i)
+""")
+    assert v.unreachable_code == []
+
+
+def test_continue_with_comment_only(v):
```

Comment:
I don't think we need the extra tests that end with comments. For the AST the comments don't even exist.